blockchain/blocks/src/tipset.rs

```diff
@@ -133,6 +134,21 @@ impl Tipset {
     pub fn weight(&self) -> &BigInt {
         self.min_ticket_block().weight()
     }
+    /// Returns true if self wins according to the filecoin tie-break rule
+    /// (FIP-0023)
+    pub fn break_weight_tie(&self, other: &Tipset) -> bool {
+        // blocks are already sorted by ticket
+        for (a, b) in self.blocks().iter().zip(other.blocks().iter()) {
```

Comment:
I believe this could be written more expressively with a call to [any](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.any). Not a big deal though.